vehicles/vehiclelighting.h

```diff
@@ -0,0 +1,46 @@
+/*
+ *     Copyright 2023 Rickard Häll   rickard.hall@ri.se
+ *     Published under GPLv3: https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ *     Code adapted from example:
+ *     https://www.ics.com/blog/gpio-programming-exploring-libgpiod-library
+ */
+
+#ifndef VEHICLELIGHTING_H
+#define VEHICLELIGHTING_H
+
+#include <QObject>
+#include <QTimer>
+#include <gpiod.h>
```

Comment:
I was wondering why raw pointers are used further below and saw that the C bindings for gpiod are used here. I would suggest to use the C++ bindings (https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/tree/bindings/cxx), if possible. In any case I would like to avoid raw pointers if there is no good reason to use them.

Reply:
C++ bindings are now used